designs/autoscaler.md

```diff
@@ -0,0 +1,515 @@
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  annotations:
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size: {{.maxCount}}
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: {{.minCount}}
+    cluster.x-k8s.io/cluster-name: {{.clusterName}}
+  name: {{.workerNodeGroupName}}
+  namespace: eksa-system
+spec:
+  clusterName: {{.clusterName}}
+  replicas: {{.count}}
+  ...
+```
+
+Note: the cluster will not have any autoscaling functionality until the Cluster Autoscaler controller is properly configured and installed in the cluster. Setting up the MachineDeployment annotations only guarantees that Cluster Autoscaler can recognize the worker node group and its autoscaling configuration. We will add a warning message in CLI logs when the autoscaling is configured in EKS-A cluster spec but autoscaler is not installed in the cluster.
+
+### Cluster Autoscaler Controller
+
+Cluster Autoscaler is a tool that automatically adjusts the size of the Kubernetes cluster. Specifically, it can:
+
+* increase the size of the cluster when there are pending pods due to insufficient resources.
+* decrease the size of the cluster when some nodes have low utilization for a significant amount of time.
+
+Just like any other application, Cluster Autoscaler can be installed as a `Deployment` in any Kubernetes cluster. In order to connect Cluster Autoscaler to Cluster API managed EKS Anywhere cluster, a user needs to configure Cluster Autoscaler to use `clusterapi` as the cloud provider, and provide the path to the kubeconfig for the target cluster to autoscale.
+
+Currently, it is impossible to have one autoscaler controller to watch multiple workload clusters with different kubeconfig at the same time, as there is no way currently to separate the pods between the workload clusters autoscaler is watching. The recommended method is to run multiple autoscalers in the management cluster with different options for how they discover the scalable resource sets.
+
+Cluster Autoscaler provides the flexibility to enable autoscaling, whether it is within the cluster, or for a remote workload cluster. Cluster Autoscaler can be:
+
+* deployed in the management cluster to autoscale the management cluster itself
+* deployed in the management cluster to autoscale a remote workload cluster
+* deployed in the workload cluster to autoscale the workload cluster itself
+
+Since EKS Anywhere follows a standard management/workload clusters setup where all the CAPI machine objects are defined in the management cluster, we choose to **deploy autoscalers in the management cluster as well — to watch and manage remote workload clusters, as default option**. In this way, we have a centralized place — management cluster, to deploy and manage all the autoscaler controllers for workload clusters. 
```

Comment:
This does not describe how EKS-A might support autoscaling of management clusters. Is that out of scope?

Reply:
we do support it. i will add a sentense to be more clear.